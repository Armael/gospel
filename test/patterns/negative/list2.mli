val f : 'a list option -> int
(*@ r = f l
    requires match l with
      | None -> false
      | Some (x :: _ as a) -> false
*)
