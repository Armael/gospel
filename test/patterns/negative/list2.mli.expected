
*******************************
********** Parsed file ********
*******************************
val f : 'a list option -> int[@@gospel
                               {| r = f l
    requires match l with
      | None -> false
      | Some (x :: _ as a) -> false
|}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Stdlib *)

(*@ open Gospelstdlib *)

val f : 'a list option -> int
(*@ r = f l
    requires ...
     *)
File "list2.mli", line 3, characters 13-83:
Error: This pattern-matching is not exhaustive.
       Here is an example of a case that is not matched:
         Some [].
